src/app/content/hooks/registerPageView.ts

```diff
@@ -0,0 +1,22 @@
+import googleAnalyticsClient from '../../../gateways/googleAnalyticsClient';
+import * as selectNavigation from '../../navigation/selectors';
+import { RouteHookBody } from '../../navigation/types';
+import { AppServices, MiddlewareAPI } from '../../types';
+import { content } from '../routes';
+
+export const hookBody: RouteHookBody<typeof content> = (services: MiddlewareAPI & AppServices) => async(action) => {
```

Comment:
i'm not sure about having prevLocation in the state, did you have a reason for doing that in redux? i was trying to suggest doing it here, by creating a cache layer like this:

```
export const hookBody: RouteHookBody<typeof content> = (services: MiddlewareAPI & AppServices) => {
  let lastTrackedLocation;

  return async(action) => {
      const state = services.getState();
      const pathname = selectNavigation.pathname(state);
      const query = selectNavigation.query(state);
      const prevPath = lastTrackedLocation?.pathname;
      const prevQuery = lastTrackedLocation?.query;

      const pathOrModalChanged =
      pathname !== prevPath || prevQuery?.modal !== query.modal;

      if (action.action !== 'REPLACE' && pathOrModalChanged) {
        lastTrackedLocation = {query, pathname};
        googleAnalyticsClient.trackPageView(pathname, query);
     }
  }
}
```

Reply:
I tried something like this initially, but this hook gets called on every locationChange so you can't cache the previous location here. Instead I cache it in routeHook and then on each locationChange it's added to the payload. I wasn't sure how else to make it accessible from within the registerPageView hook.